src/pf-dropdown/pf-dropdown.component.js

```diff
@@ -0,0 +1,164 @@
+    this._button = this.querySelector('[data-toggle="dropdown"]');
+    this._disabled = /\bdisabled/.test(this._button.className);
+
+    this._button.addEventListener('click', () => {
+      this._showDropdown();
+    });
+
+    document.addEventListener('click', (event) => {
+      if (event.target !== this && !this.contains(event.target)) {
+        this._clearDropdown();
+      }
+    });
+
+    document.addEventListener('keydown', (event) => {
+      if (!/(38|40|27|13)/.test(event.keyCode) || /input|textarea/.test(event.target.tagName)) {
+        return;
+      }
+      if (this._disabled) {
+        return;
+      }
+      let active = /\bopen/.test(this._button.parentNode.className);
+      if (active) {
+        this._keyHandler(event);
+      }
+    });
+  }
+
+  /**
+   * Called when element's attribute value has changed
+   *
+   * @param {string} attrName The attribute name that has changed
+   * @param {string} oldValue The old attribute value
+   * @param {string} newValue The new attribute value
+   */
+  attributeChangedCallback(attrName, oldValue, newValue) {
+
+  }
+
+  /**
+   * Called when an instance of the element is created
+   */
+  createdCallback() {
+
+  }
+
+  /**
+   *Toggle the dropdown
+   */
+  toggle() {
+    this._showDropdown();
+  }
+
+  /**
+   * Open the dropdown
+   *
+   */
+  _showDropdown() {
+    if (/\bdisabled/.test(this._button.className)) {
+      return;
+    }
+    let active = this._button.parentNode.classList.contains('open');
+    if (!active) {
+      this._detectTouch();
+      this.dispatchEvent(new CustomEvent('show.bs.dropdown', {}));
+      this._button.focus();
+      this._button.setAttribute('aria-expanded', 'true');
+      this._button.parentNode.classList.toggle('open');
+      this.dispatchEvent(new CustomEvent('shown.bs.dropdown', {}));
+    }
+    if (active) {
+      this._clearDropdown();
+    }
+  }
+  /**
+   * Close the dropdown
+   *
+   */
+  _clearDropdown() {
+    let backdrop = this.querySelector('.dropdown-backdrop');
+    if (backdrop) {
+      backdrop.parentNode.removeChild(backdrop);
+    }
+    this.dispatchEvent(new CustomEvent('hide.bs.dropdown', {}));
+    this._button.setAttribute('aria-expanded', 'false');
+    this._button.parentNode.classList.remove('open');
+    this.dispatchEvent(new CustomEvent('hidden.bs.dropdown', {}));
+  }
+
+  /**
+   * Support for phone browser
+   *
+   */
+  _detectTouch() {
+    if ('ontouchstart' in document.documentElement) {
+      let div = document.createElement('div');
+      div.classList.add('dropdown-backdrop');
+      this.insertBefore(div, this.querySelector('.dropdown-menu'));
+      div.addEventListener('click', this._clearDropdown());
+    }
+  }
+
+  /**
+   *Support for accessibility
+   *
+   * @param {Event} event
+   */
+  _keyHandler(event) {
+
+    event.preventDefault();
+    event.stopPropagation();
+
+    if (event.keyCode === 27 || event.keyCode === 13) {
+      this._clearDropdown();
```

Comment:
In addition, we need to handle IE, Edge, etc. as well. For example, I grabbed this code from the table's filter which uses event.keyCode or event.which.

Sorry about the indentation.

ctx._pfFilter.filterInput.on("keypress", function (e) {
      var keycode = (e.keyCode ? e.keyCode : e.which);
      if (keycode === 13) {
	e.preventDefault();
        if (this.value.trim().length > 0) {
          addFilter(dt, {
            column: ctx._pfFilter.filterColumn,
            name: ctx._pfFilter.filterName,
            value: this.value
          });
        }
        return false;
      }
      return true;
    });